support-payment-api/src/main/scala/backend/PaymentBackend.scala

```diff
@@ -0,0 +1,45 @@
+package backend
+
+import cats.data.EitherT
+import cats.implicits._
+import com.gu.acquisition.typeclasses.AcquisitionSubmissionBuilder
+import com.gu.support.acquisitions.{AcquisitionDataRow, AcquisitionsStreamService, BigQueryService}
+import com.typesafe.scalalogging.StrictLogging
+import model.DefaultThreadPool
+import model.db.ContributionData
+import services.{AnalyticsService, ContributionsStoreService}
+
+import scala.concurrent.Future
+import scala.util.control.NonFatal
+
+trait PaymentBackend extends StrictLogging {
+  val ophanService: AnalyticsService
+  val bigQueryService: BigQueryService
+  val acquisitionsStreamService: AcquisitionsStreamService
+  val databaseService: ContributionsStoreService
+
+  private def insertContributionDataIntoDatabase(contributionData: ContributionData)(implicit pool: DefaultThreadPool): EitherT[Future, BackendError, Unit] = {
+    // log so that if something goes wrong we can reconstruct the missing data from the logs
+    logger.info(s"about to insert contribution into database: $contributionData")
+    databaseService.insertContributionData(contributionData)
+      .leftMap(BackendError.fromDatabaseError)
+  }
+
+  def track[A : AcquisitionSubmissionBuilder](legacyAcquisition: A, acquisition: AcquisitionDataRow, contributionData: ContributionData)(implicit pool: DefaultThreadPool): Future[List[BackendError]] = {
```

Comment:
previously each payment backend had its own implementation for sending these events